File: 198/sol.go
func rob(nums []int) int {
    if len(nums) > 1 {
        nums[1] = max(nums[0], nums[1])
    }
    if len(nums) > 2 {
        for i := 2; i < len(nums); i++ {
            nums[i] = max(nums[i - 1], nums[i - 2] + nums[i])
        }
    }
    return nums[len(nums) - 1]
}

func max(x, y int) int {
    if x > y {
        return x
    }
    return y
}

File: 198/sol.ts
function rob(nums: number[]): number {
    if (nums.length > 1) nums[1] = Math.max(nums[0], nums[1]);
    if (nums.length > 2)
        for (let i = 2; i < nums.length; i++)
			nums[i] = Math.max(nums[i - 1], nums[i - 2] + nums[i]);
    return nums[nums.length - 1];
};
